src/dataflow/lib/iot.ts

```diff
@@ -86,15 +194,13 @@ export class IoT {
       sessionToken
     });
     this.client = connect(url);
-    this.client.subscribe(MQTT_TOPIC);
     this.client.on("message", (topic, rawMessage) => {
       const message = JSON.parse(rawMessage as any);
-      const { client: clientName, value } = message;
-      if (this.client && value) {
-        const thing = thingStore.things.get(clientName);
-        if (thing) {
-          thing.setValue(value);
-        }
+      const topicParts = topic.split("/");
```

Comment:
These hard-coded array indices seem awkward. How would you feel about a function like this:
```typescript
function parseTopic(topic: string) {
  const topicParts = topic && topic.split("/");
  return topicParts && { ownerId: topicParts[0], hubId: topicParts[2], suffix: topicParts[3] };
}
```

where clients can then use:
```typescript
const { suffix } = parseTopic(topic);
```
to extract the pieces they're interested in?

Reply:
I like it.  Going to use `leafLevel` instead of `suffix` though.  Fixed.